### PR TITLE
fix: Only confirm DB deletion when there are archived workflows. Fixes #11658

### LINKED ISSUE #11658: Deleting a workflow prompts me to also "delete from database"
### Pre-requisites

- [X] I have double-checked my configuration
- [X] I can confirm the issues exists when I tested with `:latest`
- [ ] I'd like to contribute the fix myself (see [contributing guide](https://github.com/argoproj/argo-workflows/blob/master/docs/CONTRIBUTING.md))

### What happened/what you expected to happen?

Deleting the workflow via the UI results in this popup: 
image

The warning makes no sense because we do not have workflow archiving enabled. (for what it's worth, we do have log archiving enabled, but this uses s3, not a database).

The workflow is deleted regardless of whether I press `cancel` or `ok` anyway. So from a UX perspective, it's a bit baffling. Frankly, even if I did have workflow archiving enabled - that's not a friendly error message for first-time-users.

### Version

3.5-rc1

### Paste a small workflow that reproduces the issue. We must be able to run the workflow; don't enter a workflows that uses private images.

```YAML
Any workflow.
```

### Logs from the workflow controller

```text
time="2023-08-23T10:53:32.377Z" level=info msg="Processing workflow" namespace=ci workflow=renovate-pipeline-mono-prbj4
time="2023-08-23T10:53:32.378Z" level=info msg="Task-result reconciliation" namespace=ci numObjs=0 workflow=renovate-pipeline-mono-prbj4
time="2023-08-23T10:53:32.378Z" level=info msg="node unchanged" namespace=ci nodeID=renovate-pipeline-mono-prbj4-3492216532 workflow=renovate-pipeline-mono-prbj4
time="2023-08-23T10:53:32.378Z" level=info msg="Terminating pod as part of workflow shutdown" namespace=ci podName=renovate-pipeline-mono-prbj4-renovate-3492216532 shutdownStrategy=Stop workflow=renovate-pipeline-mono-prbj4
time="2023-08-23T10:53:32.378Z" level=info msg="node renovate-pipeline-mono-prbj4-3492216532 phase Running -> Failed" namespace=ci workflow=renovate-pipeline-mono-prbj4
time="2023-08-23T10:53:32.378Z" level=info msg="node renovate-pipeline-mono-prbj4-3492216532 message: workflow shutdown with strategy:  Stop" namespace=ci workflow=renovate-pipeline-mono-prbj4
time="2023-08-23T10:53:32.378Z" level=info msg="node renovate-pipeline-mono-prbj4-3492216532 finished: 2023-08-23 10:53:32.378636191 +0000 UTC" namespace=ci workflow=renovate-pipeline-mono-prbj4
time="2023-08-23T10:53:32.379Z" level=info msg="Stopped with strategy 'Stop'" namespace=ci workflow=renovate-pipeline-mono-prbj4
time="2023-08-23T10:53:32.379Z" level=info msg="node renovate-pipeline-mono-prbj4-1573759850 phase Running -> Failed" namespace=ci workflow=renovate-pipeline-mono-prbj4
time="2023-08-23T10:53:32.379Z" level=info msg="node renovate-pipeline-mono-prbj4-1573759850 message: Stopped with strategy 'Stop'" namespace=ci workflow=renovate-pipeline-mono-prbj4
time="2023-08-23T10:53:32.379Z" level=info msg="node renovate-pipeline-mono-prbj4-1573759850 finished: 2023-08-23 10:53:32.379189863 +0000 UTC" namespace=ci workflow=renovate-pipeline-mono-prbj4
time="2023-08-23T10:53:32.379Z" level=info msg="Outbound nodes of renovate-pipeline-mono-prbj4-2344736112 set to [renovate-pipeline-mono-prbj4-3492216532]" namespace=ci workflow=renovate-pipeline-mono-prbj4
time="2023-08-23T10:53:32.379Z" level=info msg="node renovate-pipeline-mono-prbj4-2344736112 phase Running -> Failed" namespace=ci workflow=renovate-pipeline-mono-prbj4
time="2023-08-23T10:53:32.379Z" level=info msg="node renovate-pipeline-mono-prbj4-2344736112 finished: 2023-08-23 10:53:32.379311141 +0000 UTC" namespace=ci workflow=renovate-pipeline-mono-prbj4
time="2023-08-23T10:53:32.379Z" level=info msg="Stopped with strategy 'Stop'" namespace=ci workflow=renovate-pipeline-mono-prbj4
time="2023-08-23T10:53:32.379Z" level=info msg="node renovate-pipeline-mono-prbj4 phase Running -> Failed" namespace=ci workflow=renovate-pipeline-mono-prbj4
time="2023-08-23T10:53:32.379Z" level=info msg="node renovate-pipeline-mono-prbj4 message: Stopped with strategy 'Stop'" namespace=ci workflow=renovate-pipeline-mono-prbj4
time="2023-08-23T10:53:32.379Z" level=info msg="node renovate-pipeline-mono-prbj4 finished: 2023-08-23 10:53:32.379483979 +0000 UTC" namespace=ci workflow=renovate-pipeline-mono-prbj4
time="2023-08-23T10:53:32.379Z" level=info msg="TaskSet Reconciliation" namespace=ci workflow=renovate-pipeline-mono-prbj4
time="2023-08-23T10:53:32.379Z" level=info msg=reconcileAgentPod namespace=ci workflow=renovate-pipeline-mono-prbj4
time="2023-08-23T10:53:32.379Z" level=info msg="Updated phase Running -> Failed" namespace=ci workflow=renovate-pipeline-mono-prbj4
time="2023-08-23T10:53:32.379Z" level=info msg="Updated message  -> Stopped with strategy 'Stop'" namespace=ci workflow=renovate-pipeline-mono-prbj4
time="2023-08-23T10:53:32.379Z" level=info msg="Marking workflow completed" namespace=ci workflow=renovate-pipeline-mono-prbj4
time="2023-08-23T10:53:32.384Z" level=info msg="cleaning up pod" action=terminateContainers key=ci/renovate-pipeline-mono-prbj4-renovate-3492216532/terminateContainers
time="2023-08-23T10:53:32.384Z" level=info msg="https://172.20.0.1:443/api/v1/namespaces/ci/pods/renovate-pipeline-mono-prbj4-renovate-3492216532/exec?command=%2Fvar%2Frun%2Fargo%2Fargoexec&command=kill&command=15&command=1&container=main&stderr=true&stdout=true&tty=false"
time="2023-08-23T10:53:32.385Z" level=info msg="cleaning up pod" action=deletePod key=ci/renovate-pipeline-mono-prbj4-1340600742-agent/deletePod
time="2023-08-23T10:53:32.394Z" level=info msg="Workflow update successful" namespace=ci phase=Failed resourceVersion=138747944 workflow=renovate-pipeline-mono-prbj4
time="2023-08-23T10:53:32.395Z" level=info msg="Queueing Failed workflow ci/renovate-pipeline-mono-prbj4 for delete in 1h0m0s due to TTL"
time="2023-08-23T10:53:32.407Z" level=info msg="cleaning up pod" action=labelPodCompleted key=ci/renovate-pipeline-mono-prbj4-renovate-3492216532/labelPodCompleted
time="2023-08-23T10:53:32.538Z" level=info msg="signaled container" container=main error="<nil>" namespace=ci pod=renovate-pipeline-mono-prbj4-renovate-3492216532 stderr= stdout="killing 1 with terminated\n"
time="2023-08-23T10:53:32.538Z" level=info msg="https://172.20.0.1:443/api/v1/namespaces/ci/pods/renovate-pipeline-mono-prbj4-renovate-3492216532/exec?command=%2Fvar%2Frun%2Fargo%2Fargoexec&command=kill&command=15&command=1&container=wait&stderr=true&stdout=true&tty=false"
time="2023-08-23T10:53:32.628Z" level=info msg="signaled container" container=wait error="<nil>" namespace=ci pod=renovate-pipeline-mono-prbj4-renovate-3492216532 stderr= stdout="killing 1 with terminated\n"
time="2023-08-23T10:53:36.967Z" level=info msg="Processing ci/renovate-pipeline-mono" cronWorkflow=ci/renovate-pipeline-mono
time="2023-08-23T10:53:36.969Z" level=info msg="CronWorkflow ci/renovate-pipeline-mono added" cronWorkflow=ci/renovate-pipeline-mono
time="2023-08-23T10:54:02.629Z" level=info msg="cleaning up pod" action=killContainers key=ci/renovate-pipeline-mono-prbj4-renovate-3492216532/killContainers
time="2023-08-23T10:57:28.921Z" level=info msg="Queueing Failed workflow ci/renovate-pipeline-mono-prbj4 for delete in 56m4s due to TTL"
time="2023-08-23T10:57:29.065Z" level=info msg="Queueing Failed workflow ci/renovate-pipeline-mono-prbj4 for delete in 56m3s due to TTL"
```

### Logs from in your workflow's wait container

```text
time="2023-08-23T10:43:42.160Z" level=info msg="Starting Workflow Executor" version=v3.5.0-rc1
time="2023-08-23T10:43:42.163Z" level=info msg="Using executor retry strategy" Duration=1s Factor=1.6 Jitter=0.5 Steps=5
time="2023-08-23T10:43:42.163Z" level=info msg="Executor initialized" deadline="2023-08-23 11:35:20 +0000 UTC" includeScriptOutput=false namespace=ci podName=renovate-pipeline-mono-prbj4-renovate-3492216532 template="{\"name\":\"renovate\",\"inputs\":{},\"outputs\":{},\"nodeSelector\":{\
time="2023-08-23T10:43:42.163Z" level=info msg="Starting deadline monitor"
time="2023-08-23T10:48:42.164Z" level=info msg="Alloc=6974 TotalAlloc=13886 Sys=30829 NumGC=6 Goroutines=7"
time="2023-08-23T10:53:32.622Z" level=info msg="Deadline monitor stopped"
time="2023-08-23T10:53:32.622Z" level=info msg="stopping progress monitor (context done)" error="context canceled"
time="2023-08-23T10:53:33.222Z" level=warning msg="Non-transient error: context canceled"
time="2023-08-23T10:53:33.222Z" level=info msg="Main container completed" error="context canceled"
time="2023-08-23T10:53:33.222Z" level=info msg="No Script output reference in workflow. Capturing script output ignored"
time="2023-08-23T10:53:33.222Z" level=info msg="No output parameters"
time="2023-08-23T10:53:33.222Z" level=info msg="No output artifacts"
time="2023-08-23T10:53:33.222Z" level=info msg="S3 Save path: /tmp/argo/outputs/logs/main.log, key: my-artifacts/2023/08/23/renovate-pipeline-mono-prbj4/renovate-pipeline-mono-prbj4-renovate-3492216532/main.log"
time="2023-08-23T10:53:33.222Z" level=info msg="Creating minio client using AWS SDK credentials"
time="2023-08-23T10:53:33.329Z" level=info msg="Saving file to s3" bucket=pipekit-argo-workflows endpoint=s3.amazonaws.com key=my-artifacts/2023/08/23/renovate-pipeline-mono-prbj4/renovate-pipeline-mono-prbj4-renovate-3492216532/main.log path=/tmp/argo/outputs/logs/main.log
time="2023-08-23T10:53:33.530Z" level=info msg="Save artifact" artifactName=main-logs duration=307.244472ms error="<nil>" key=my-artifacts/2023/08/23/renovate-pipeline-mono-prbj4/renovate-pipeline-mono-prbj4-renovate-3492216532/main.log
time="2023-08-23T10:53:33.530Z" level=info msg="not deleting local artifact" localArtPath=/tmp/argo/outputs/logs/main.log
time="2023-08-23T10:53:33.530Z" level=info msg="Successfully saved file: /tmp/argo/outputs/logs/main.log"
time="2023-08-23T10:53:33.558Z" level=info msg="Alloc=11295 TotalAlloc=18297 Sys=35181 NumGC=8 Goroutines=9"
```

### PR DESCRIPTION
**Pull request by terrytangyuan**
Fixes #11658

### CODE REVIEW

File: ui/src/app/workflows/components/workflows-toolbar/workflows-toolbar.tsx [hunk 1]
```diff
@@ -47,11 +47,19 @@ export class WorkflowsToolbar extends React.Component<WorkflowsToolbarProps, {}>
...
             if (title === 'DELETE') {
-                deleteArchived = confirm('Would you also want to delete them in the database?');
+                let hasArchived = false;
+                this.props.selectedWorkflows.forEach((wf: Workflow) => {
+                    if (!hasArchived && isArchivedWorkflow(wf)) {
+                        hasArchived = true;
+                    }
+                });
+                if (hasArchived) {
+                    deleteArchived = confirm('Would you also want to delete them in the database?');
```
Comment (tico24) [inline ui/src/app/workflows/components/workflows-toolbar/workflows-toolbar.tsx:61]: I'm still not convinced this is the best wording. Consider `Would you also like to delete them from the Workflow Archive database?`

Comment (terrytangyuan) [reply ui/src/app/workflows/components/workflows-toolbar/workflows-toolbar.tsx:61]: Are there any other databases?

Comment (terrytangyuan) [reply ui/src/app/workflows/components/workflows-toolbar/workflows-toolbar.tsx:61]: That seems like implementation detail

Comment (tico24) [reply ui/src/app/workflows/components/workflows-toolbar/workflows-toolbar.tsx:61]: Depends if your workflow envokes a database or not. If I was a first-time user, or a user that wasn't involved in the setup of Workflows, that error would confuse the heck out of me.

**Review by agilgur5 — APPROVED** [review]

File: ui/src/app/workflows/components/workflows-toolbar/workflows-toolbar.tsx [hunk 2]
```diff
@@ -47,11 +47,19 @@ export class WorkflowsToolbar extends React.Component<WorkflowsToolbarProps, {}>
...
-                deleteArchived = confirm('Would you also want to delete them in the database?');
+                let hasArchived = false;
+                this.props.selectedWorkflows.forEach((wf: Workflow) => {
+                    if (!hasArchived && isArchivedWorkflow(wf)) {
+                        hasArchived = true;
+                    }
+                });
+                if (hasArchived) {
+                    deleteArchived = confirm('Would you also want to delete them from the Archived Workflows database?');
+                }
```
Comment (agilgur5) [inline ui/src/app/workflows/components/workflows-toolbar/workflows-toolbar.tsx:62]:
```suggestion
                for (const entry of this.props.selectedWorkflows) {
                    if (isArchivedWorkflow(entry[1])) {
                        deleteArchived = confirm('Would you also want to delete them in the database?');
                        break
                    }
                }
```

ok that actually ended up being a decent simplification / reduction

File: ui/src/app/workflows/components/workflows-toolbar/workflows-toolbar.tsx [hunk 3]
```diff
@@ -47,11 +47,19 @@ export class WorkflowsToolbar extends React.Component<WorkflowsToolbarProps, {}>
...
 
     private performActionOnSelectedWorkflows(ctx: any, title: string, action: WorkflowOperationAction): Promise<any> {
-        const confirmed = confirm(`Are you sure you want to ${title.toLowerCase()} all selected workflows?`);
         let deleteArchived = false;
+        const confirmed = confirm(`Are you sure you want to ${title.toLowerCase()} all selected workflows?`);
         if (confirmed) {
             if (title === 'DELETE') {
-                deleteArchived = confirm('Would you also want to delete them in the database?');
+                let hasArchived = false;
+                this.props.selectedWorkflows.forEach((wf: Workflow) => {
```
[the remark this answers is not in the document]
Comment (agilgur5) [reply ui/src/app/workflows/components/workflows-toolbar/workflows-toolbar.tsx:55]: See [below comment](https://github.com/argoproj/argo-workflows/pull/11659#discussion_r1303414376)

[the remark this answers is not in the document]
Comment (terrytangyuan) [reply ui/src/app/workflows/components/workflows-toolbar/workflows-toolbar.tsx]: Thanks!

**Review by rbreeze — APPROVED** [review]

**Pull request merged**